### DISCUSSION TITLE
s3 cloudfront as disk

### DISCUSSION THREAD

**Discussion by johnnywo** [awcodes/filament-curator#220] [Q&A] [answered]
Hi!
I am able to successfully create an image in admin Media which is successfully uploaded to s3, 
and which is also displayed nicely in admin /edit page with correct cloudfront url.
I am also able to display the image on any custom page with:
```
@php
   $imageModel = \Awcodes\Curator\Models\Media::find($image);
@endphp
<img src="{{ Storage::disk('s3')->url($imageModel->path) }}" alt="My Photo" />
```
But: I am not able to use curator glider:
`<x-curator-glider :media="$image" class="object-cover object-center rounded" alt="{{ $title }}" />`

And I am not able to display the images in admin on the list page as thumbnails. They aren't showing up .. 
League\Glide\Filesystem\FileNotFoundException: Could not find the image `public/media/564b028f-af3b-48d3-be60-a7c5f14d9ef7.webp`. in file /Users/johnnywo/Sites/wte/vendor/league/glide/src/Server.php on line 629

Any tips appreciated!
Thanks

**Comment by awcodes** [comment, marked as answer]
There's a PR that I need to review that should solve some of this for you.

But I'm not 100% sure that Glide will even work with external urls, since there's not an actual file on your server for it to process.

**Reply by johnnywo** [reply]
thanks, okay!
Just helped me with your answer as I was thinking about it - my primary goal for this task was to save original images to s3. 
If the cached images are getting served from my local storage is secondary. If using cloudfront as cdn is secondary. It just works now with those settings from a serviceprovider: 
```
Curator::disk('s3')
            ->glideServer(ServerFactory::create([
                'source' => Storage::disk('s3')->getDriver(),
                'cache' => storage_path('app'),
                'driver' => 'gd',
                'response' => new LaravelResponseFactory(app('request')),
                'cache_path_prefix' => '.cache',
                'max_image_size' => 2000 * 2000,
            ]));
```
